bundles/persistence/org.openhab.persistence.influxdb/java/org/openhab/persistence/influxdb/internal/InfluxDBPersistenceService.java

```diff
@@ -445,16 +458,22 @@ private Object stateToObject(State state) {
    */
   private String stateToString(State state) {
     String value;
-    if (state instanceof PercentType) {
-      value = ((PercentType) state).toBigDecimal().toString();
-    } else if (state instanceof DateTimeType) {
-      value = String.valueOf(((DateTimeType) state).getCalendar().getTime().getTime());
-    } else if (state instanceof DecimalType) {
-      value = ((DecimalType) state).toBigDecimal().toString();
+    if (state instanceof DecimalType) {
+    	value = ((DecimalType) state).toBigDecimal().toString();
+    } else if (state instanceof PercentType) {
+    	value = ((PercentType) state).toBigDecimal().toString();
```

Comment:
PercentType is a DecimalType and for DecimalType calling toString() is the right thing.